### PR TITLE
Discord Bee

### PR DESCRIPTION
**Pull request by CalmBit**
Adds the discord bee! (Tackling #235 )

Unfortunately, it requires a bit of wrangling (a user has to create a discord app, create a bot on that app, get its API token, and use that to seed the bot) - this is all unavoidable by Discord's design, however. Everything else is fluid!

### CODE REVIEW

File: bees/discordbee/discordbee.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Authors:
+ *      CalmBit <calmbit@posto.net>
+ *      Christian Muehlhaeuser <muesli@gmail.com>
+ */
+
+// Package discordbee is
```
Comment (muesli) [inline bees/discordbee/discordbee.go]: Time to finish that description :smile:

File: bees/discordbee/discordbee.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+package discordbee
+
+import (
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/muesli/beehive/bees"
+)
+
+// DiscordBee is
```
Comment (muesli) [inline bees/discordbee/discordbee.go]: Time to finish that description :smile:

File: bees/discordbee/discordbee.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+	select {
+	case <-mod.SigChan:
+		return
+	}
+}
+
+func (mod *DiscordBee) isReady(s *discordgo.Session, event *discordgo.Ready) {
+	mod.Logf("Is ready")
+	s.UpdateStatus(0, "with bees")
```
Comment (muesli) [inline bees/discordbee/discordbee.go]: I'm a bit confused by this. Are we required to set some kind of status to make this work, or was this just part of a test?

File: bees/discordbee/discordbeefactory.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+func (factory *DiscordBeeFactory) Options() []bees.BeeOptionDescriptor {
+	opts := []bees.BeeOptionDescriptor{
+		{
+			Name:        "api_token",
+			Description: "The Discord API token for your bot",
+			Type:        "string",
+			Mandatory:   true,
+		},
+		{
+			Name:        "trigger_phrase",
```
Comment (muesli) [inline bees/discordbee/discordbeefactory.go]: That sounds like something that should be handled by a user-created filter. I would simply emit all message events and let the user decide when to trigger an action, e.g.:

```
{{test HasPrefix .contents ".sometrigger"}}
```

File: bees/discordbee/discordbeefactory.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	return opts
+}
+
+// Events describes the available events provided by this Bee.
+func (factory *DiscordBeeFactory) Events() []bees.EventDescriptor {
+	events := []bees.EventDescriptor{
+		{
+			Namespace:   factory.Name(),
+			Name:        "message_received",
```
Comment (muesli) [inline bees/discordbee/discordbeefactory.go]: Unless you want to re-use the "receive" or "send" keywords in events & actions (for receiving/sending some other than messages), I would suggest simply calling this event `message` and the action `send`.

File: bees/discordbee/discordbee.go (hunk 3 above)
Comment (CalmBit) [reply bees/discordbee/discordbee.go]: Updating the status is just a decorative thing - it should actually probably be a configuration option!

Comment (muesli) [reply bees/discordbee/discordbee.go]: I see! Maybe we should expose it as an action then? `set_status` or such.

Comment (CalmBit) [reply bees/discordbee/discordbee.go]: Ah, that's not a bad idea! Adding that to the list of things to do :)

File: bees/discordbee/discordbee.go (hunk 1 above)
Comment (CalmBit) [reply bees/discordbee/discordbee.go]: I mean, to be fair, in the existential sense, is it _really_ wrong? 🤔 

In all seriousness - big whoops. Will be fixed!

File: bees/discordbee/discordbeefactory.go (hunk 4 above)
Comment (CalmBit) [reply bees/discordbee/discordbeefactory.go]: The only thing I'd be worried about in relation to that is the general dragnet approach that would entail - printing every message to the log seems kinda surveillance-ee to me.

File: bees/discordbee/discordbee.go [hunk 6]
```diff
@@ -20,17 +20,17 @@
  *      Christian Muehlhaeuser <muesli@gmail.com>
  */
 
-// Package discordbee is
+// Package discordbee is a bee for sending and recieving messages with Discord
```
Comment (muesli) [inline bees/discordbee/discordbee.go]: Just a heads up, typo in `receiving` here.

File: bees/discordbee/discordbee.go [hunk 7]
```diff
@@ -20,17 +20,17 @@
...
 
 import (
-	"strings"
-
 	"github.com/bwmarrin/discordgo"
 	"github.com/muesli/beehive/bees"
 )
 
-// DiscordBee is
+// DiscordBee is a bee for sending and recieving messages with Discord
```
Comment (muesli) [inline bees/discordbee/discordbee.go]: Copy/paste typo :smile:

File: bees/discordbee/discordbeefactory.go [hunk 8]
```diff
@@ -90,8 +84,8 @@ func (factory *DiscordBeeFactory) Events() []bees.EventDescriptor {
 	events := []bees.EventDescriptor{
 		{
 			Namespace:   factory.Name(),
-			Name:        "message_received",
-			Description: "is triggered when a message has been recieved with the trigger word",
+			Name:        "message",
+			Description: "is triggered when a message has been recieved",
```
Comment (muesli) [inline bees/discordbee/discordbeefactory.go]: Copy/paste typo :smile:

Comment (CalmBit) [reply bees/discordbee/discordbeefactory.go]: This one is me just being dumb!

File: .vscode/settings.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (muesli) [inline .vscode/settings.json]: Oops, I guess that just slipped in by accident.

Comment (CalmBit) [reply .vscode/settings.json]: Damn. I've really gotta commit that to the .gitignore soon...

**Comment by gingerbeardman** [comment]
A step by step guide for this would be awesome!

I have created the discord app and bot and have my token, and i've created my bees and chains but something is obviously missing.

**Pull request merged**